geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/distributed/WaitUntilFlushedFunction.java

```diff
@@ -85,8 +72,10 @@ public void execute(FunctionContext context) {
       }
 
     } else {
-      throw new IllegalArgumentException(
+      IllegalStateException illegalStateException = new IllegalStateException(
           "The AEQ does not exist for the index " + indexName + " region " + region.getFullPath());
+      logger.error(illegalStateException.getMessage());
```

Comment:
I think it would be better here to just throw the IllegalStateException. That makes it clear that we intend the exception to propagate back to the caller.

Based on what I've seen from my testing, calling resultSender.lastResult(new IllegalStateException()) will behave the same as throwing the exception: on the remote side ResultCollector.getResult throws an exception. However, I think that is kind of weird, whereas it makes sense that if you throw an exception in the function it will propagate back as an exception on the remote side.